src/bin/main.rs

```diff
@@ -163,13 +162,14 @@ async fn main() {
         work_done_progress_params: WorkDoneProgressParams {
             work_done_token: Some(lsp_types::NumberOrString::Number(2)),
         },
+        ..Default::default()
     };
 
     let (tx, rx) = oneshot::channel();
     lang_server
         .send_request("initialize", &json!(init_params), |result| {
             println!("received response {:?}", result);
-            tx.send(result);
+            tx.send(result).unwrap();
```

Comment:
this is totally okay to do for now (as we have no error handling), but try to use expect here instead of unwrap since with expect you at the very least get a good error message when it crashes while running.